pkg/pipelinerunmetrics/metrics_test.go

```diff
@@ -369,14 +373,108 @@ func TestRecordRunningPipelineRunsCount(t *testing.T) {
 		t.Fatalf("NewRecorder: %v", err)
 	}
 
-	if err := metrics.RunningPipelineRuns(informer.Lister()); err != nil {
+	if err := metrics.RunningPipelineRuns(prInformer.Lister(), trInformer.Lister()); err != nil {
 		t.Errorf("RunningPipelineRuns: %v", err)
 	}
 	metricstest.CheckLastValueData(t, "running_pipelineruns_count", map[string]string{}, 1)
 }
 
+func TestRecordRunningPipelineRunsThrottledCounts(t *testing.T) {
+	unregisterMetrics()
+	ctx, _ := ttesting.SetupFakeContext(t)
+	prInformer := fakepipelineruninformer.Get(ctx)
+	trInformer := faketaskruninformer.Get(ctx)
+
+	newTaskRun := func(status corev1.ConditionStatus, reason string) []*v1beta1.TaskRun {
+		retArray := []*v1beta1.TaskRun{}
+		// add 2 taskruns to make sure we don't count a given exceed condition more than once
+		retArray = append(retArray, &v1beta1.TaskRun{
+			ObjectMeta: metav1.ObjectMeta{Name: names.SimpleNameGenerator.RestrictLengthWithRandomSuffix("taskrun-"), Namespace: metav1.NamespaceDefault},
+			Status: v1beta1.TaskRunStatus{
+				Status: duckv1.Status{
+					Conditions: duckv1.Conditions{{
+						Type:   apis.ConditionSucceeded,
+						Status: status,
+						Reason: reason,
+					}},
+				},
+			},
+		})
+		retArray = append(retArray, &v1beta1.TaskRun{
+			ObjectMeta: metav1.ObjectMeta{Name: names.SimpleNameGenerator.RestrictLengthWithRandomSuffix("taskrun-"), Namespace: metav1.NamespaceDefault},
+			Status: v1beta1.TaskRunStatus{
+				Status: duckv1.Status{
+					Conditions: duckv1.Conditions{{
+						Type:   apis.ConditionSucceeded,
+						Status: status,
+						Reason: reason,
+					}},
+				},
+			},
+		})
+		return retArray
+	}
+	newPipelineRun := func(status corev1.ConditionStatus, reasons []string) *v1beta1.PipelineRun {
+		pr := &v1beta1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{Name: names.SimpleNameGenerator.RestrictLengthWithRandomSuffix("pipelinerun-"), Namespace: metav1.NamespaceDefault},
+			Status: v1beta1.PipelineRunStatus{
+				Status: duckv1.Status{
+					Conditions: duckv1.Conditions{{
+						Type:   apis.ConditionSucceeded,
+						Status: status,
+					}},
+				},
+			},
+		}
+		kidRefs := []v1beta1.ChildStatusReference{}
+		for _, reason := range reasons {
+			trs := newTaskRun(status, reason)
+			for _, tr := range trs {
+				// NOTE: while GetIndexer().Add(obj) works well in these unit tests for List(), for
+				// trLister.TaskRuns(trNamespace).Get(trName) to work correctly we need to use GetStore() instead.
+				// see AddToInformer in test/controller.go for an analogous solution
```

Comment:
why not use `SeedTestData` or `AddToInformer` from test/controller.go instead? (here and in the taskrun tests)

Reply:
Yeah I was trying to stay as close as possible to the patterns previously established in this test, like what I saw in `TestRecordRunningPipelineRunsCount` 

But sure, I can tinker with say using `SeedTestData` in my new test here and see where that lands.